remoulade/state/backends/redis.py

```diff
@@ -26,25 +26,42 @@ def __init__(self, *, namespace="remoulade-state", encoder=None, client=None, ur
         super().__init__(namespace=namespace, encoder=encoder)
         if url:
             parameters["connection_pool"] = redis.ConnectionPool.from_url(url)
+        parameters["decode_responses"] = True
         self.client = client or redis.Redis(**parameters)
 
     def get_state(self, message_id):
         key = self._build_message_key(message_id)
-        data = self.client.get(key)
-        if data:
+        data = self.client.hgetall(key)
+        if len(data) == 0:
```

Comment:
You can write it more compact. use Not instead of len == 0 it make things more clear (less to read).
```
        if not data:
            return None
        return self._parse_state(data)
```